source/renderer/app/components/wallet/WalletSendForm.js

```diff
@@ -696,28 +703,64 @@ export default class WalletSendForm extends Component<Props, State> {
       !this.hasAvailableAssets ? styles.disabled : null,
       'primary',
     ]);
+
+    const receiverFieldClasses = classNames([
+      styles.receiverInput,
+      this.isAddressFromSameWallet() ? styles.sameRecieverInput : null,
+    ]);
+
     const minAdaRequiredTooltip = selectedAssetFingerprints.length
       ? messages.minAdaRequiredWithAssetTooltip
       : messages.minAdaRequiredWithNoAssetTooltip;
 
     return (
       <div className={styles.fieldsContainer}>
-        <div className={styles.receiverInput}>
-          <Input
-            {...receiverField.bind()}
-            ref={(field) => {
-              this.addFocusableField(field);
-            }}
-            className="receiver"
-            error={receiverField.error}
-            onChange={(value) => {
-              receiverField.onChange(value || '');
-              this.setState({
-                isResetButtonDisabled: false,
-              });
-            }}
-            onKeyPress={this.handleSubmitOnEnter}
-          />
+        <div className={receiverFieldClasses}>
+          {this.isAddressFromSameWallet() ? (
+            <PopOver
+              content={intl.formatMessage(messages.sameWalletLabel, {
+                walletName,
+              })}
+              contentClassName={styles.sameWalletTooltipContent}
+              themeVariables={{
+                '--rp-pop-over-bg-color':
+                  'var(--rp-password-input-warning-score-color)',
+              }}
+              placement="bottom"
+            >
+              <Input
+                {...receiverField.bind()}
+                ref={(field) => {
+                  this.addFocusableField(field);
+                }}
+                className="receiver"
+                error={receiverField.error}
+                onChange={(value) => {
+                  receiverField.onChange(value || '');
+                  this.setState({
+                    isResetButtonDisabled: false,
+                  });
+                }}
+                onKeyPress={this.handleSubmitOnEnter}
+              />
+            </PopOver>
+          ) : (
+            <Input
+              {...receiverField.bind()}
+              ref={(field) => {
+                this.addFocusableField(field);
+              }}
+              className="receiver"
+              error={receiverField.error}
+              onChange={(value) => {
+                receiverField.onChange(value || '');
+                this.setState({
+                  isResetButtonDisabled: false,
+                });
+              }}
+              onKeyPress={this.handleSubmitOnEnter}
+            />
+          )}
```

Comment:
@aleksandardjordjeviciohk let's refactor the duplication of the `Input` component here - we could either save it into a variable above the return and re-use it for the popover condition logic or maybe even better: use a conditional [disabled](https://github.com/atomiks/tippyjs-react#disabled-boolean) boolean flag on the `PopOver` (haven't tried that yet but would be good to test if that works)

Reply:
Fixed

Reply:
Refactored @DominikGuzei